### ISSUE TITLE
Replace exfat-utils with exfatprogs

### ISSUE THREAD

**Issue by Vogtinator** [yast/yast-storage-ng#1214]
exfat-utils was apparently replaced by exfatprogs, and SLE doesn't even have the former.

https://github.com/yast/yast-storage-ng/blob/e505276388da1fd6fb1bb49de7321e4dd63629f1/src/lib/y2storage/storage_feature.rb#L78

**Comment by ancorgs** [comment]
As far as I can see in a quick check:

- Leap 15.2 included both exfatprogs and exfat-utils
- SLE-15-SP2 included none of those two packages (no exfat support at all)

- Leap 15.3 includes exfatprogs but NOT exfat-utils
- SLE-15-SP3 includes exfatprogs but NOT exfat-utils

- Currently Tumbleweed includes both exfatprogs and exfat-utils

So switching to exfatprogs looks like a safe bet. Having said so, do you have any pointer to the explanation about why exfat-utils was dropped in 15.3?

**Comment by Vogtinator** [comment]
> Having said so, do you have any pointer to the explanation about why exfat-utils was dropped in 15.3?

The explanation is that SP3 already had exfatprogs. So there's not much substance behind that reasoning...

**Comment by Vogtinator** [comment]
According to https://lore.kernel.org/lkml/004701d6194c$0d238990$276a9cb0$@samsung.com/, they're both the same project, but looking at git, exfat-utils is something different. At least I'm even more confused now.

**Comment by ancorgs** [comment]
Both packages provide binaries in `/usr/sbin` with the basic names you would expect (`fsck.exfat`, `mkfs.exfat`, `exfatlabel`) and, thus, the packages conflict with each other.

But there is an important difference. The package exfat-utils also provides symlinks at `/sbin` for those binaries, while the package exfatprogs only provides the `/usr/sbin/` paths. Why is that important? Because as far as I can see, libstorage-ng expects those files in `/sbin`.
https://github.com/openSUSE/libstorage-ng/blob/master/storage/Utils/StorageDefines.h#L138

So before changing this in yast2-storage-ng, we should adapt libstorage-ng to use the binaries at `/usr/sbin`, which should be compatible with both packages.

@aschnell Am I right?

**Comment by aschnell** [comment]
Yes, using /usr/sbin should work for both packages. But users setting mkfs option might face compatibility problems since the tools have different options.

**Comment by ancorgs** [comment]
Now tracked at https://bugzilla.suse.com/show_bug.cgi?id=1184489 and (for the YaST team) at https://trello.com/c/N7gz5WP0/

**Comment by rokejulianlockhart** [comment]
Considering that https://bugzilla.suse.com/show_bug.cgi?id=1184489 is RESOLVED FIXED, should this be closed?

**Comment by ancorgs** [comment]
This is only fixed in the libstorage-ng side (which was the critical part). But YaST would still try to add `exfat-utils` to the selection of packages (unsuccessfully). That still needs to be improved.
